File: Java/civitas/src/juegoTexto/VistaTextual.java
package juegoTexto;

import civitas.CivitasJuego;
import civitas.Diario;
import civitas.OperacionesJuego;
import civitas.SalidasCarcel;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;
import civitas.Casilla;
import civitas.CivitasJuego;
import civitas.Jugador;
import civitas.OperacionesJuego;
import civitas.Respuestas;
import civitas.SalidasCarcel;
import civitas.TituloPropiedad;

class VistaTextual {
  
  CivitasJuego juegoModel; 
  int iGestion=-1;
  int iPropiedad=-1;
  private static String separador = "=====================";
  
  private Scanner in;
  
  VistaTextual () {
    in = new Scanner (System.in);
  }
  
  void mostrarEstado(String estado) {
    System.out.println (estado);
  }
              
  void pausa() {
    System.out.print ("Pulsa la tecla ENTER para avanzar");
    in.nextLine();
  }

  int leeEntero (int max, String msg1, String msg2) {
    Boolean ok;
    String cadena;
    int numero = -1;
    do {
      System.out.print (msg1);
      cadena = in.nextLine();
      try {  
        numero = Integer.parseInt(cadena);
        ok = true;
      } catch (NumberFormatException e) { // No se ha introducido un entero
        System.out.println (msg2);
        ok = false;  
      }
      if (ok && (numero < 0 || numero >= max)) {
        System.out.println (msg2);
        ok = false;
      }
    } while (!ok);

    return numero;
  }

  int menu (String titulo, ArrayList<String> lista) {
    String tab = "  ";
    int opcion;
    System.out.println (titulo);
    for (int i = 0; i < lista.size(); i++) {
      System.out.println (tab+i+"-"+lista.get(i));
    }

    opcion = leeEntero(lista.size(),
                          "\n"+tab+"Elige una opción: ",
                          tab+"Valor erróneo");
    return opcion;
  }

  SalidasCarcel salirCarcel() {
    int opcion = menu ("Elige la forma para intentar salir de la carcel",
    new ArrayList<> (Arrays.asList("Pagando","Tirando el dado")));
    return (SalidasCarcel.values()[opcion]);
  }

  Respuestas comprar() {
    int opcion = menu ("Elige si desea comprar la calle en la que ha caido",
    new ArrayList<> (Arrays.asList("NO","SI")));
    return (Respuestas.values()[opcion]);
  }

  void gestionar () {
      int opcion2 = 0;
      int opcion = menu ("Elige la accion que desea realizar", 
      new ArrayList<> (Arrays.asList("VENDER", "HIPOTECAR", "CANCELAR_HIPOTECA", "CONSTRUIR_CASA", "CONSTRUIR_HOTEL", "TERMINAR")));
      ArrayList<String> propiedades = new ArrayList<>();
      if(5 == opcion){
        iGestion=5;
      } 
      else {
        for(int i=0; i<juegoModel.getJugadorActual().getPropiedades().size(); i++){
            propiedades.add(juegoModel.getJugadorActual().getPropiedades().get(i).getNombre());
        }
        opcion2 = menu ("Elige la propiedad sobre la que realizar la operacion", propiedades);
        iGestion = opcion;
        iPropiedad = opcion2; 
      }
      
      
  }

    public int getiGestion() {
        return iGestion;
    }

    public int getiPropiedad() {
        return iPropiedad;
    }


  void mostrarSiguienteOperacion(OperacionesJuego operacion) {
      System.out.println("La siguiente operacion que se va a realizar es " + operacion.toString());
  }


  void mostrarEventos() {
      while(Diario.getInstance().eventosPendientes()){
          System.out.println(Diario.getInstance().leerEvento());
      }
  }
  
  public void setCivitasJuego(CivitasJuego civitas){ 
        juegoModel=civitas;

    }
  
  void actualizarVista(){
      System.out.println(juegoModel.getJugadorActual().toString());
      System.out.println(juegoModel.getCasillaActual());
  } 
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/vista_textual.rb
#encoding:utf-8
require_relative 'operaciones_juego'
require 'io/console'
require_relative 'civitasjuego'
require_relative 'tablero'
require_relative 'salidas_carcel'
require_relative 'respuestas'
require_relative 'gestiones_inmobiliarias'


module Civitas

  class Vista_textual

    attr_accessor :juegoModel
    def mostrar_estado(estado)
      puts estado
    end

    
    def pausa
      print "Pulsa una tecla"
      STDIN.getch
      print "\n"
    end

    def lee_entero(max,msg1,msg2)
      ok = false
      begin
        print msg1
        cadena = gets.chomp
        begin
          if (cadena =~ /\A\d+\Z/)
            numero = cadena.to_i
            ok = true
          else
            raise IOError
          end
        rescue IOError
          puts msg2
        end
        if (ok)
          if (numero >= max)
            ok = false
          end
        end
      end while (!ok)

      return numero
    end



    def menu(titulo,lista)
      tab = "  "
      puts titulo
      index = 0
      lista.each { |l|
        puts tab+index.to_s+"-"+l
        index += 1
      }

      opcion = lee_entero(lista.length,
                          "\n"+tab+"Elige una opción: ",
                          tab+"Valor erróneo")
      return opcion
    end

    
    def comprar()
      opciones = Array.new()
      respuesta = [Respuestas::SI,Respuestas::NO]
      opciones << "SI"
      opciones << "NO"
      opcion = menu("Elige si comprar o no la propiedad", opciones)
      return respuesta[opcion]
    end

    def gestionar()
      gestiones=[Gestiones_inmobiliarias::VENDER,Gestiones_inmobiliarias::HIPOTECAR,Gestiones_inmobiliarias::CANCELAR_HIPOTECA, 
       Gestiones_inmobiliarias::CONSTRUIR_CASA,Gestiones_inmobiliarias::CONSTRUIR_HOTEL,Gestiones_inmobiliarias::TERMINAR]
      opciones = Array.new()
      opciones << "VENDER"
      opciones << "HIPOTECAR"
      opciones << "CANCELAR_HIPOTECA"
      opciones << "CONSTRUIR_CASA"
      opciones << "CONSTRUIR_HOTEL"
      opciones << "TERMINAR"
       opcion = menu("Elige la acccion que desea realizar", opciones)
      propiedades = Array.new()
      if(5==opcion)
        @gestion = gestiones[opcion]
      else
        for i in 0..(@juegoModel.jugadores.at(@juegoModel.indice_jugador_actual).propiedades.size()-1)
          propiedades << @juegoModel.jugadores.at(@juegoModel.indice_jugador_actual).propiedades.at(i).nombre
        end
        opcion2 = menu("Elige la propiedad sobre la que realizar la operacion", propiedades)
        @gestion = gestiones[opcion]
        @propiedad=opcion2
      end
      
      
    end

    def get_gestion()
      return @gestion
    end

    def get_propiedad
     return @propiedad
    end

    def mostrarSiguienteOperacion(operacion)
      puts "La siguiente operacion que se va a realizar es " + operacion.to_s()
    end

    def mostrarEventos
      while(Diario.instance.eventos_pendientes())
        puts Diario.instance.leer_evento()
      end
    end

    def setCivitasJuego(civitas)
         @juegoModel=civitas
    end
    
    def salir_carcel()
      opciones = Array.new()
      salida = [Salidas_carcel::PAGANDO,Salidas_carcel::TIRANDO]
      opciones << "Pagando"
      opciones << "Tirando el dado"
      opcion = menu("Elige forma de salir de la carcel", opciones)
      return salida[opcion]
    end
    
    

    def actualizarVista()
      jugadoractual = @juegoModel.jugadores.at(@juegoModel.indice_jugador_actual)
      puts jugadoractual.to_sj()
      puts @juegoModel.get_casilla_actual().to_sc()
      
    end

    
  end

end
